lib/api/unittest/CDataFrameAnalyzerFeatureImportanceTest.cc

```diff
@@ -142,10 +142,15 @@ void setupMultiClassClassificationData(const TStrVec& fieldNames,
     maths::CPRNG::CXorOShiro128Plus rng;
     std::uniform_real_distribution<double> u01;
     int numberFeatures{static_cast<int>(weights.size())};
-    TDoubleVec w{weights};
     int numberClasses{static_cast<int>(classes.size())};
+    TDoubleVec storage(numberClasses * numberFeatures);
+    for (int i = 0; i < numberClasses; ++i) {
```

Comment:
Should it be `numberFeatures`?  Or if not on this line somewhere.  There is an inconsistency somewhere between lines 146 and 149.

Reply:
Good point. In fact, because we resize the vector (we're okay), the extra weights are just zero, so it's harmless but not intended. I'll correct this. Also Eigen expects the data to be column major and I'd intended to use the same weights for each class so I've switch that as well.

Reply:
Fixed in [240937b](https://github.com/elastic/ml-cpp/pull/1086/commits/240937bd915bf2adb6bd4a69d4ca5b3338093c03).